### ISSUE TITLE
Chrome browser on External Selenium server - ERROR

### ISSUE THREAD

**Issue by softwareklinic** [sitespeedio/sitespeed.io#1306]
2016-11-02 16:44:54] NoSuchElementError: no such element: Unable to locate element: {"method":"css selector","selector":"*[id="IDToken1"]"}
  (Session info: chrome=54.0.2840.71)
  (Driver info: chromedriver=2.25.426923 (0390b88869384d6eb0d5d09729679f934aab9eed),platform=Windows NT 6.3.9600 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 12 milliseconds

**Comment by softwareklinic** [comment]
Need some urgent help...

**Comment by soulgalore** [comment]
Does it work locally? To be able to help we need the script and URLs so we can reproduce the problem but if it's a Selenium script issue you can probably find it in the Selenium docs http://seleniumhq.github.io/selenium/docs/api/javascript/index.html and fix it yourself?

**Comment by softwareklinic** [comment]
It was working on the server with old chrome driver… 2.22 --- but after that it stopped last week – so updated the driver… and since then.. anything where I need to find element – doesn’t work.

Also, doesn’t work on local as well… same issue.. so looks like with the new chrome driver --- we are unable to find element.

**Comment by softwareklinic** [comment]
It was working on the server with old chrome driver… 2.22 --- but after that it stopped last week – so updated the driver… and since then.. anything where I need to find element – doesn’t work.

Also, doesn’t work on local as well… same issue.. so looks like with the new chrome driver --- we are unable to find element.

**Comment by softwareklinic** [comment]
I have OLD sitespeed alpha 5 working on server... and latest master code running on local... and none of my scripts with find element are working anymore... so if u can check locally would be helpful...

**Comment by soulgalore** [comment]
Ok, I can test tonight (in 3-4 hours), we have updated Chromedriver to the latest. Can you give me a one liner when you try to get an element?

**Comment by soulgalore** [comment]
Did check with master and latest docker 4.0.3 with https://github.com/sitespeedio/browsertime/blob/master/test/prepostscripts/preLoginExample.js and it worked fine. You are sure the page hasn't changed?

**Comment by softwareklinic** [comment]
What i did was used another external selenium server -- with chrome driver 2.22 and chrome 53.x --- (old) --- which worked just fine... 

So, it did break with chrome driver 2.25 and chrome 54.x --- so i can try to get it from master again and try it out.

**Comment by softwareklinic** [comment]
got latest from master and running from local...

[2016-11-02 14:55:40] NoSuchElementError: no such element: Unable to locate element: {"method":"css selector","selector":"*[id="IDToken1"]"}
  (Session info: chrome=54.0.2840.71)
  (Driver info: chromedriver=2.24.417412 (ac882d3ce7c0d99292439bf3405780058fcca0a6),platform=Mac OS X 10.11.6 x86_64)

**Comment by softwareklinic** [comment]
here is my full script - no changes to script --- works on 1 set of chromedriver+browser and doesnt work on the latest driver+browser

'use strict';

module.exports = {
  run(context) {
    return context.runWithDriver((driver) => {
      return driver.get('https://www.verizonwireless.com/')
        .then(() => {

 var webdriver = context.webdriver;
var By = webdriver.By;
          var userName = 'xxxxxxxx';
          var sq = 'xxxxxxxx';
          var password = 'xxxxxxxx';
          // NAVIGATING TO HOME PAGE
          var loginInput = driver.findElement(By.id('IDToken1'));
          loginInput.click();
          loginInput.clear();
          loginInput.sendKeys(userName);

          driver.findElement(By.xpath("(//button[@type='submit'])[4]")).click();

          // NAVIGATING TO THE SECURITY QUESTION ANSWER PAGE
          var securequestion = driver.findElement(By.id('IDToken1'));
          securequestion.click();
          securequestion.clear();
          securequestion.sendKeys(sq);
    	  driver.findElement(By.css("button.o-red-button")).click();	

		  // NAVIGATING TO THE PASSWORD PAGE
          var passwordinput = driver.findElement(webdriver.By.id('IDToken2'));
          passwordinput.click();
          passwordinput.clear();
          passwordinput.sendKeys(password);
          driver.findElement(By.css("button.o-red-button")).click();
          // NAVIGATES TO THE HOME PAGE - AS SIGNED IN
        });
    })
  }
};

**Comment by softwareklinic** [comment]
One more thing.... it works for the 1st Set of inputs... username.... but once it clicks on the submit button and 2nd page loads ----it just crashes right there with element not found---even before displaying the page being loaded...

**Comment by soulgalore** [comment]
So this should be a Chrome issue then?

**Comment by softwareklinic** [comment]
What's ur chrome and driver version

Sent from my iPhone

On Nov 2, 2016, at 3:51 PM, Peter Hedenskog <notifications@github.com<mailto:notifications@github.com>> wrote:


So this should be a Chrome issue then?

-

**Comment by soulgalore** [comment]
It's Google Chrome 54.0.2840.71 in Docker and the same local and Chromedriver 2.25.0 (the driver is installed automatically using npm).

**Comment by softwareklinic** [comment]
Since.I'm.using external selenium.server -.could it be selenium.jar?

Sent from my iPhone

On Nov 2, 2016, at 4:15 PM, Peter Hedenskog <notifications@github.com<mailto:notifications@github.com>> wrote:


It's Google Chrome 54.0.2840.71 in Docker and the same local and Chromedriver 2.25.0 (the driver is installed automatically using npm).

-

**Comment by abhagupta** [comment]
This is similar issue that I stumbled upon few days ago.  Basically your browser is not opening or, even if its opening, the selenium does not recognize it as completely open and times out. The error `no such element: Unable to locate element` is just a result of first action  that selenium is trying to do on the page which does not show anything.  In my case,  I upgraded my sitespeed version which updated the selenium version and Chrome to 54. Things have been normal since then.  

@softwareklinic  how are you using sitespeed.io in your tests?  May be you are missing an option for browser in your command.

**Comment by soulgalore** [comment]
@abhagupta so then if I understand correctly we are missing testing that things really works before we start to do the tests. We check the URL when we analyze a URL but there's no way to check the HTTP status code in Selenium, we should add this to the FAQ as a known problem.

**Comment by softwareklinic** [comment]
Since.I'm.behind.corporate.firewall and for all practical reasons I'm.unable to use the sitespeed dockerized version of selenium I'm using  external selenium server on Windows.with chrome.and Firefox

Sent from my iPhone

On Nov 3, 2016, at 11:57 AM, Abha Gupta <notifications@github.com<mailto:notifications@github.com>> wrote:


This is similar issue that I stumbled upon few days ago. Basically your browser is not opening or, even if its opening, the selenium does not recognize it as completely open and times out. The error no such element: Unable to locate element is just a result of first action that selenium is trying to do on the page which does not show anything. In my case, I upgraded my sitespeed version which updated the selenium version and Chrome to 54. Things have been normal since then.

@softwareklinic<https://github.com/softwareklinic> how are you using sitespeed.io<http://sitespeed.io> in your tests? May be you are missing an option for browser in your command.

-
